tests/test_2025_12_10_mainnet_v3.py

```diff
@@ -0,0 +1,796 @@
+STAKING_ROUTER = "0xFdDf38947aFB03C621C71b06C9C70bce73f12999"
+ACL = "0x9895F0F17cc1d1891b6f18ee0b483B6f221b37Bb"
+NODE_OPERATORS_REGISTRY = "0x55032650b14df07b85bF18A3a3eC8E0Af2e028d5"
+SIMPLE_DVT = "0xaE7B191A31f627b4eB1d4DaC64eaB9976995b433"
+ORACLE_DAEMON_CONFIG = "0xbf05A929c3D7885a6aeAd833a992dA6E5ac23b09"
+CSM_ACCOUNTING = "0x4d72BFF1BeaC69925F8Bd12526a39BAAb069e5Da"
+OLD_BURNER = "0xD15a672319Cf0352560eE76d9e89eAB0889046D3"
+LIDO_APP_ID = "0x3ca7c3e38968823ccb4c78ea688df41356f182ae1d159e4ee608d30d68cef320"
+
+# New Lido V3 addresses - TODO fix after mainnet deployment
+VAULT_HUB = "0xdcC04F506E24495E9F2599A7b214522647363669"
+ACCOUNTING = "0xc9158c756D2a510eaC85792C847798a30dE20D46"
+ACCOUNTING_IMPL = "0x1bE2Ee7D32e3F9C4ecd5b6BfF69306ACb8Ab239e"
+OPERATOR_GRID = "0x79e2685C1DD4756AC709a6CeE7C6cC960128B031"
+LAZY_ORACLE = "0x4fA3c917BB8f8CD9d056C5DDF0a38bd1834c43F9"
+PREDEPOSIT_GUARANTEE = "0x7B49b203A100E326B84886dCC0e2c426f9b8cbBd"
+VAULTS_ADAPTER = "0x8cDA09f41970A8f4416b6bA4696a2f09a6080c76"
+GATE_SEAL_V3 = "0x9c2D30177DB12334998EB554f5d4E6dD44458167"
+LIDO_IMPL = "0xD0b9826e0EAf6dE91CE7A6783Cd6fd137ae422Ec"
+UPGRADE_TEMPLATE = "0x2AE2847a77a1d24100be2163AfdC49B06DC808E4"
+LIDO_LOCATOR_IMPL = "0x26329a3D4cF2F89923b5c4C14a25A2485cD01aA2"
+ACCOUNTING_ORACLE_IMPL = "0xb2295820F5286BE40c2da8102eB2cDB24aD608Be"
+RESEAL_MANAGER = "0x7914b5a1539b97Bd0bbd155757F25FD79A522d24"
+BURNER = "0xD140f4f3C515E1a328F6804C5426d9e8b883ED50"
+
+ALTER_TIERS_IN_OPERATOR_GRID_FACTORY = "0x25AB9D07356E8a3F95a5905f597c93CD8F31990b"
+REGISTER_GROUPS_IN_OPERATOR_GRID_FACTORY = "0x9Ba3E4aDDe415943A831b97F9f7B8b842052b709"
+REGISTER_TIERS_IN_OPERATOR_GRID_FACTORY = "0x663cE8Aa1ded537Dc319529e71DE6BAb2E7D0747"
+SET_JAIL_STATUS_IN_OPERATOR_GRID_FACTORY = "0x88D32aABa5B6A972D82E55D26B212eBeca07C983"
+SET_LIABILITY_SHARES_TARGET_IN_VAULT_HUB_FACTORY = "0xf61601787E84d89c30911e5A48d9CA630eC47044"
+SOCIALIZE_BAD_DEBT_IN_VAULT_HUB_FACTORY = "0xD60671a489948641954736A0e7272137b3A335CE"
+FORCE_VALIDATOR_EXITS_IN_VAULT_HUB_FACTORY = "0x6D0F542591eF4f8ebA8c77a11dc3676D9E9F7e66"
+UPDATE_GROUPS_SHARE_LIMIT_IN_OPERATOR_GRID_FACTORY = "0xAEDB8D15197984D39152A2814e0BdCDAEED5462d"
+UPDATE_VAULTS_FEES_IN_OPERATOR_GRID_FACTORY = "0x7DCf0746ff2F77A36ceC8E318b59dc8a8A5c066e"
+
+UTC14 = 60 * 60 * 14
+UTC23 = 60 * 60 * 23
+SLASHING_RESERVE_SHIFT = 8192
+
+EXPECTED_VOTE_ID = 194
+EXPECTED_DG_PROPOSAL_ID = 6
+EXPECTED_VOTE_EVENTS_COUNT = 10
+EXPECTED_DG_EVENTS_FROM_AGENT = 17
+EXPECTED_DG_EVENTS_COUNT = 18
+IPFS_DESCRIPTION_HASH = "bafkreic4xuaowfowt7faxnngnzynv7biuo7guv4s4jrngngjzzxyz3up2i"
+
+
+# ============================================================================
+# ============================== Helper functions ============================
+# ============================================================================
+
+def validate_proxy_upgrade_event(event: EventDict, implementation: str, emitted_by: Optional[str] = None, events_chain: Optional[list] = None):
```

Comment:
if this is a generic validator function that can be used for future votings - consider moving it to the utils/test/event_validators
please check wheter this validator function already exists in utils/test/event_validators